src/agents/agent.py

```diff
@@ -63,7 +64,16 @@ def search_queries(self, queries: list, project_name: str) -> dict:
         results = {}
         
         knowledge_base = KnowledgeBase()
-        bing_search = BingSearch()
+        user_search_engine = Config().get_search_engine()
+
+        if user_search_engine == "DDGS":
+            search_engine = DuckDuckGoSearch()
+        else:
+            """
+            Set default search engine to Bing
+            """
+            search_engine = BingSearch()
```

Comment:
**Security Issue**: The dynamic selection of the search engine based on user preference introduces a risk where an attacker could potentially manipulate the application to use a less secure or compromised search engine. This could lead to information disclosure, injection attacks, or redirection to malicious sites. <br> **Fix**: Validate the user's selected search engine against a predefined list of allowed search engines. This ensures that only approved search engines can be used, mitigating the risk of using an insecure or compromised search engine. <br> **Code Suggestion**: 
 ```
 +ALLOWED_SEARCH_ENGINES = ['Bing', 'DDGS']
 
  def validate_search_engine(engine):
      if engine not in ALLOWED_SEARCH_ENGINES:
          raise ValueError(f'{engine} is not a supported search engine.')
 
  class Config:
      def get_search_engine(self):
 +        user_engine = self.config['USER_PREFS']['SEARCH_ENGINE']
 +        validate_search_engine(user_engine)
          return user_engine

 ```